src/behavior.ts

```diff
@@ -26,174 +26,193 @@
-        this.bindMouseEventsToClearCatcher(clearCatcher);
-        this.bindMouseEventsToLegend(legend);
-
-        this.bindKeyboardEventsToDataPoints(selection);
+    private onSelectCallback(selectionIds?: ISelectionId[]){
+        this.applySelectionStateToData(selectionIds);
+        this.renderSelection();
     }
 
-    private bindMouseEventsToDataPoints(selection): void {
-        selection.on("click", (event: PointerEvent, dataPoint: HierarchyRectangularNode<SunburstDataPoint>) => {
-            const isMultiSelection: boolean = event.ctrlKey || event.metaKey || event.shiftKey;
-            this.selectionManager.select(dataPoint.data.identity, isMultiSelection);
-
-            this.renderSelection();
-            event.stopPropagation();
-        });
+    private applySelectionStateToData(selectionIds?: ISelectionId[]): void{
+        const selectedIds: ISelectionId[] = <ISelectionId[]>this.selectionManager.getSelectionIds();
+        this.setSelectedToDataPoints(this.dataPoints, selectionIds || selectedIds);
+        this.setSelectedToDataPoints(this.legendDataPoints, selectionIds || selectedIds);
+    }
 
-        selection.on("contextmenu", (event: PointerEvent, dataPoint: HierarchyRectangularNode<SunburstDataPoint>) => {
-            this.selectionManager.showContextMenu(dataPoint.data.identity, {
-                x: event.clientX,
-                y: event.clientY
+    private setSelectedToDataPoints(dataPoints: LegendDataPoint[] | HierarchyRectangularNode<SunburstDataPoint>[], ids: ISelectionId[]): void{
+        dataPoints.forEach((dataPoint: LegendDataPoint| HierarchyRectangularNode<SunburstDataPoint>) => {
+            const data : SunburstDataPoint | LegendDataPoint = (dataPoint as HierarchyRectangularNode<SunburstDataPoint>)?.data || (dataPoint as LegendDataPoint);
+            data.selected = false;
+            ids.forEach((selectedId: ISelectionId) => {
+                if (selectedId.includes(<ISelectionId>data.identity)) {
+                    data.selected = true;
+                }
             });
-            event.preventDefault();
-            event.stopPropagation();
         });
     }
 
-    private bindMouseEventsToClearCatcher(clearCatcher): void{
-        clearCatcher.on("click", () => {
-            this.selectionManager.clear();
-            this.renderSelection();
-        });
-
-        clearCatcher.on("contextmenu", (event: PointerEvent) => {
-            this.selectionManager.showContextMenu({}, {
-                x: event.clientX,
-                y: event.clientY
-            });
+    private bindContextMenuEvent(elements: d3Selection<any>): void {
+        elements.on("contextmenu", (event: PointerEvent, dataPoint: HierarchyRectangularNode<SunburstDataPoint> | LegendDataPoint | undefined) => {
+            const data : SunburstDataPoint | LegendDataPoint = (dataPoint as HierarchyRectangularNode<SunburstDataPoint>)?.data || (dataPoint as LegendDataPoint);
```

Comment:
Looks complicated, what do  you think about moving it to a separate function (as it's used in several places)
Just like this
``` typescript
private castDataFromPoint(dataPoint: HierarchyRectangularNode<SunburstDataPoint> | LegendDataPoint | undefined) {
   return (dataPoint as HierarchyRectangularNode<SunburstDataPoint>)?.data || (dataPoint as LegendDataPoint);
}
```